strawberry/schema/schema_converter.py

```diff
@@ -530,45 +532,86 @@ def _strawberry_info_from_graphql(info: GraphQLResolveInfo) -> Info:
                 _field=field,
             )
 
-        def _get_result(_source: Any, info: Info, **kwargs):
-            field_args, field_kwargs = _get_arguments(
-                source=_source, info=info, kwargs=kwargs
-            )
-
+        def _get_result(
+            _source: Any,
+            info: Info,
+            field_args: List[Any],
+            field_kwargs: Dict[str, Any],
+        ):
             return field.get_result(
                 _source, info=info, args=field_args, kwargs=field_kwargs
             )
 
         def wrap_field_extensions() -> Callable[..., Any]:
             """Wrap the provided field resolver with the middleware."""
 
-            if not field.extensions:
-                return _get_result
-
             for extension in field.extensions:
                 extension.apply(field)
 
             extension_functions = build_field_extension_resolvers(field)
-            return reduce(
-                lambda chained_fns, next_fn: partial(next_fn, chained_fns),
-                extension_functions,
-                _get_result,
-            )
+
+            def extension_resolver(
+                _source: Any,
+                info: Info,
+                **kwargs,
+            ):
+                # parse field arguments into Strawberry input types and convert
+                # field names to Python equivalents
+                field_args, field_kwargs = _get_arguments(
+                    source=_source, info=info, kwargs=kwargs
+                )
+
+                resolver_requested_info = False
+                if "info" in field_kwargs:
+                    resolver_requested_info = True
+                    # remove info from field_kwargs because we're passing it
+                    # explicitly to the extensions
+                    field_kwargs.pop("info")
+
+                # `_get_result` expects `field_args` and `field_kwargs` as
+                # separate arguments so we have to wrap the function so that we
+                # can pass them in
+                def wrapped_get_result(_source, info, **kwargs):
+                    # if the resolver function requested the info object info
+                    # then put it back in the kwargs dictionary
+                    if resolver_requested_info:
+                        kwargs["info"] = info
+
+                    return _get_result(
+                        _source, info, field_args=field_args, field_kwargs=kwargs
+                    )
+
+                # combine all the extension resolvers
+                return reduce(
+                    lambda chained_fn, next_fn: partial(next_fn, chained_fn),
+                    extension_functions,
+                    wrapped_get_result,
+                )(_source, info, **field_kwargs)
```

Comment:
Have you tested if this has any performance implications? Doing it this way means we have to do the reduce on _every_ resolver call. It should be a small impact per-resolver, but I feel like GQL-core caches these for a reason.

Reply:
I haven't because it should be operating on an empty array if there are no extensions. Will check though.

Reply:
@erikwrede 

Before:
```
tests/benchmarks/test_execute.py 😃 😃 😃


--------------------------------------------------------------------------------- benchmark: 3 tests ---------------------------------------------------------------------------------
Name (time in ms)         Min                Max               Mean            StdDev             Median               IQR            Outliers       OPS            Rounds  Iterations
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
test_execute[25]       2.9184 (1.0)      27.2476 (2.71)      3.0810 (1.0)      1.5843 (5.97)      2.9609 (1.0)      0.0349 (1.0)          1;22  324.5737 (1.0)         235           1
test_execute[100]      8.6431 (2.96)     10.0608 (1.0)       8.7955 (2.85)     0.2652 (1.0)       8.7056 (2.94)     0.0928 (2.66)        11;12  113.6950 (0.35)        107           1
test_execute[250]     19.9964 (6.85)     42.8958 (4.26)     20.7761 (6.74)     3.2831 (12.38)    20.1603 (6.81)     0.2760 (7.92)          1;6   48.1322 (0.15)         48           1
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

```

After:
```
tests/benchmarks/test_execute.py 😃 😃 😃


--------------------------------------------------------------------------------- benchmark: 3 tests ---------------------------------------------------------------------------------
Name (time in ms)         Min                Max               Mean            StdDev             Median               IQR            Outliers       OPS            Rounds  Iterations
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
test_execute[25]       2.9457 (1.0)      26.3224 (2.85)      3.1145 (1.0)      1.5349 (19.78)     2.9959 (1.0)      0.0451 (1.0)          1;20  321.0760 (1.0)         231           1
test_execute[100]      8.7342 (2.97)      9.2507 (1.0)       8.8250 (2.83)     0.0776 (1.0)       8.8078 (2.94)     0.0596 (1.32)         17;9  113.3145 (0.35)        109           1
test_execute[250]     20.3858 (6.92)     41.6573 (4.50)     20.9648 (6.73)     3.0857 (39.77)    20.4810 (6.84)     0.1032 (2.29)          1;6   47.6991 (0.15)         47           1
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

```

It doesn't seem to make a significant difference.

Reply:
thanks for checking! ☺️